python/cudf/cudf/utils/_numba.py

```diff
@@ -7,6 +7,19 @@
 
 from numba import config as numba_config
 
+try:
+    from pynvjitlink.patch import (
+        patch_numba_linker as patch_numba_linker_cuda_12,
+    )
+except ImportError:
+
+    def patch_numba_linker_cuda_12():
```

Comment:
Do we expect that future major versions of CUDA will also use pynvjitlink? Do we want the naming of these functions to reflect "nvjitlink" rather than "cuda_12"?
```suggestion
    def patch_numba_linker_pynvjitlink():
```